### ISSUE TITLE
Extraneous factors of 2 in KL divergence?

### ISSUE THREAD

**Issue by asmello** [fastforwardlabs/vae-tf#3]
I'm not 100% sure I haven't missed something, but there's something odd in lines **183-184**:

``` python
return -0.5 * tf.reduce_sum(1 + 2 * log_sigma - mu**2 -
                                        tf.exp(2 * log_sigma), 1)
```

What are those factors of `2 *` doing there? They're not in any other VAE implementation I've seen, and not in any explicit derivation I could find. They don't even agree with the note just above:

``` python
# = -0.5 * (1 + log(sigma**2) - mu**2 - sigma**2)
```

Which also seems to be missing the exponential and sum.

**Comment by meereeum** [comment]
the two equations are equivalent.

ignoring the mu term for a minute..

``` python
>>> s = 47

>>> 2 * np.log(s) - np.exp(2*np.log(s))
-2201.2997047965791

>>> np.log(s**2) - s**2
-2201.29970479658
```

(log arithmetic!)

implementations sometimes differ in whether the encoder learns the variance (`sigma_sq`) or some "variant" of the variance (e.g. `log_sigma` or `log_sigma_sq`). you can avoid some numerical instability with logs / extra compute step by directly learning `log_sigma`... but any of these will work (as long as the KL loss is set up to correspond).

the summation term arises from summing over all examples in the minibatch.

See equation 24 (all the way at the bottom) in the [original VAE paper](https://arxiv.org/abs/1312.6114) for where this closed-form expression for KL loss comes from in the first place.

**Comment by asmello** [comment]
I think the actual equation is inside annex B, page 11. I had seen it before, which was what triggered me to find this implementation strange. But I get it now. I thought `log_sigma` was actually `log_sigma_sq`; I should've taken a clue from the name. Thanks.

Still, I suggest you make this more clear in your explanation in the blog post, or in the commented line above. It might have been just me, but maybe someone else would benefit from the clarification.
